tests/tests.c

```diff
@@ -2009,6 +2010,85 @@ bool sirtest_pluginloader(void) {
     return PRINT_RESULT_RETURN(pass);
 }
 
+bool sirtest_stringutils(void) {
+    INIT(si, SIRL_ALL, 0, 0, 0);
+    bool pass = si_init;
+
+    char str[] = "Kneel  \f \n  before  \t \r \v  Zod!?";
+
+    TEST_MSG_0(WHITEB("--- valid string utility usage ---"));
+
+    _sir_eqland(pass, _sir_strsqueeze(str) && 0 == strncmp(str, "Kneel before Zod!?", 18));
+    TEST_MSG("_sir_strsqueeze:            '%s'", str);
+
+    _sir_eqland(pass, _sir_strremove(str, "!") && 0 == strncmp(str, "Kneel before Zod?", 17));
+    TEST_MSG("_sir_strremove(\"!\"):        '%s'", str);
+
+    _sir_eqland(pass, _sir_strreplace(str, '?', '.') && 0 == strncmp(str, "Kneel before Zod.", 17));
+    TEST_MSG("_sir_strreplace(\"?\", \".\"):  '%s'", str);
+
+    _sir_eqland(pass, 1 == _sir_strcreplace(str, '.', '!', 1) && 0 == strncmp(str, "Kneel before Zod!", 17));
+    TEST_MSG("_sir_strcreplace(\".\", \"!\"): '%s'", str);
+
+    _sir_eqland(pass, _sir_strredact(str, "e", '*') && 0 == strncmp(str, "Kn**l b*for* Zod!", 17));
+    TEST_MSG("_sir_strredact(\"e\", \"*\"):   '%s'", str);
+
+    _sir_eqland(pass, _sir_strredact(str, "X", 'Y') && 0 == strncmp(str, "Kn**l b*for* Zod!", 17));
+    TEST_MSG("_sir_strredact(\"X\", \"Y\"):   '%s'", str);
+
+    PASSFAIL_MSG(pass, "\t--- valid string utility usage: %s ---\n\n", PRN_PASS(pass));
+
+    TEST_MSG_0(WHITEB("--- invalid string utility usage - NULL pointer ---"));
+
+    TEST_MSG_0("_sir_strsqueeze:  NULL pointer");
+    _sir_eqland(pass, !_sir_strsqueeze(NULL));
+
+    TEST_MSG_0("_sir_strremove:   NULL pointer");
+    _sir_eqland(pass, !_sir_strremove(NULL, "sub"));
+
+    TEST_MSG_0("_sir_strreplace:  NULL pointer");
+    _sir_eqland(pass, !_sir_strreplace(NULL, 'c', 'n'));
+
+    TEST_MSG_0("_sir_strcreplace: NULL pointer");
+    _sir_eqland(pass, !_sir_strcreplace(NULL, 'c', 'n', -1));
+
+    TEST_MSG_0("_sir_strredact:   NULL pointer");
+    _sir_eqland(pass, !_sir_strredact(NULL, "s", '*'));
+
+    PASSFAIL_MSG(pass, "\t--- invalid string utility usage - NULL pointer: %s ---\n\n", PRN_PASS(pass));
+
+    TEST_MSG_0(WHITEB("--- invalid string utility usage - bad parameters ---"));
+
+    TEST_MSG_0("_sir_strremove:   bad parameter \"sub\"");
```

Comment:
are you putting the extra spaces here and other spots on purpose? it's cool if so, just not sure if it's a typo.

Reply:
![image](https://github.com/aremmell/libsir/assets/61629094/c76f6419-1f22-4d1f-ab81-f6b79a4714bd)

I could not include the spacing ...